### PR TITLE
Use the DOM parser in the RSS Downloader contrib

### PR DESCRIPTION
**Pull request by xymus**
`rss_downloader` now uses the DOM XML parser instead of regex to read RSS files. It is also easier to customize its behavior from the config class.

This PR also adds a few services to the DOM parser and fix a few bugs.

---

This PR is based on #1448 with fixed line endings.

### CODE REVIEW

File: lib/dom/dom.nit [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	# var code = """
+	# <?xml version="1.0" encoding="us-ascii"?>
+	# <animal>
+	#     <cat/>
+	#     <tiger>This is a white tiger!</tiger>
+	#     <cat/>
+	# </animal>"""
+	#
+	# var xml = code.to_xml
+	# assert xml["animal"].first["tiger"].first.as(XMLStartTag).data == "This is a white tiger!"
```
Comment (privat) [inline lib/dom/dom.nit]: I do not like the `as(XMLStartTag)`. No alternative proposal however.

**Comment by privat** [comment]
+1

File: lib/dom/dom.nit (hunk 1 above)
Comment (xymus) [inline lib/dom/dom.nit]: I don't like it either...

**Comment by R4PaSs** [comment]
:shipit:

**Comment by privat** [comment]
please rebase because conflicts with the merged version of #1448

**Comment by xymus** [comment]
Rebased on master.

**Pull request merged**